array_api_strict/tests/test_array_object.py

```diff
@@ -410,6 +411,78 @@ def _matmul_array_vals():
                     x.__imatmul__(y)
 
 
+@pytest.mark.parametrize(
+    "op",
+    [
+        op for op, dtypes in binary_op_dtypes.items()
+        if dtypes not in ("real numeric", "floating-point")
+    ],
+)
+def test_binary_operators_vs_numpy_int(op):
+    """np.int64 is not a subclass of int and must be disallowed"""
+    a = asarray(1)
+    i64 = np.int64(1)
+    with pytest.raises(TypeError, match="Expected Array or Python scalar"):
+        getattr(a, op)(i64)
```

Comment:
Note to self: this test basically tests the status quo, the behavior is the same on main.